bundler/lib/dependabot/bundler/native_helpers.rb

```diff
@@ -1,10 +1,35 @@
 # frozen_string_literal: true
 
+require "dependabot/shared_helpers"
+
 module Dependabot
   module Bundler
     module NativeHelpers
-      def self.helper_path
-        "bundle exec ruby #{File.join(native_helpers_root, 'run.rb')}"
+      def self.run_bundler_subprocess(function:, args:, bundler_version:)
+        SharedHelpers.run_helper_subprocess(
+          command: helper_path(bundler_version: bundler_version),
+          function: function,
+          args: args,
+          # Kernel#spawn: clear environment variables except specified by env
+          unsetenv_others: true,
+          env: {
+            # Bundler will pick the matching installed major version
+            "BUNDLER_VERSION" => bundler_version,
+            # Required to find the ruby bin
+            "PATH" => ENV["PATH"],
+            # # Requried to create tmp directories in a writeable folder
+            "HOME" => ENV["HOME"],
+            # Required to git clone to a writeable folder
+            "GEM_HOME" => ENV["GEM_HOME"],
+            # Required by git fetch
+            "SSH_AUTH_SOCK" => ENV["SSH_AUTH_SOCK"]
```

Comment:
Oh we probably need for forward some env from the runner, e.g.

SSL_CERT_FILE
http_proxy
HTTP_PROXY
https_proxy
HTTPS_PROXY